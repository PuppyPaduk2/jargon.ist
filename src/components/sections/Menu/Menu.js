// @flow

import React from 'react';
import Link from 'gatsby-link';

import './Menu.scss';

const Menu = () => (
  <nav className="c-menu">
    <Link className="c-menu__link" to="/dizin" activeClassName="is-active">
      Dizin
    </Link>

    <Link className="c-menu__link" to="/konular" activeClassName="is-active">
      Konular
    </Link>

    <a
      href="https://github.com/alpcanaydin/jargon.ist"
      target="_blank"
      rel="noopener noreferrer"
      className="c-button c-button--primary"
    >
      KatkÄ±da Bulun
    </a>
  </nav>
);

export default Menu;
