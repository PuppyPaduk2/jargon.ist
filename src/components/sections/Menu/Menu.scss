@import '../../../styles/abstracts/index';

.c-menu {
  align-items: center;
  display: flex;
  flex-direction: row;
  margin-left: auto;

  &__link {
    color: $color-white;
    font-size: $g-font-size-small;
    margin-right: gap();

    &::after {
      background-color: $primary-color;
      content: '';
      display: block;
      height: 2px;
      opacity: 0;
      transition: opacity 0.4s;
      width: 100%;
    }

    &:hover,
    &.is-active {
      color: $primary-color;
      text-decoration: none;

      // stylelint-disable-next-line max-nesting-depth
      &::after {
        opacity: 1;
      }
    }
  }
}
