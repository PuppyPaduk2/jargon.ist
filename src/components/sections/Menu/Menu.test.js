// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Menu from './Menu';

it('renders correctly', () => {
  const wrapper = shallow(<Menu />);

  expect(wrapper).toMatchSnapshot();
});
