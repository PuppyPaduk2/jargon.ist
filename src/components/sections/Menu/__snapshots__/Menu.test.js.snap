// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="c-menu"
>
  <GatsbyLink
    activeClassName="is-active"
    className="c-menu__link"
    to="/dizin"
  >
    Dizin
  </GatsbyLink>
  <GatsbyLink
    activeClassName="is-active"
    className="c-menu__link"
    to="/konular"
  >
    Konular
  </GatsbyLink>
  <a
    className="c-button c-button--primary"
    href="https://github.com/alpcanaydin/jargon.ist"
    rel="noopener noreferrer"
    target="_blank"
  >
    KatkÄ±da Bulun
  </a>
</nav>
`;
